packages/core/src/agent/didcomm/versions/v1/indy/DIDCommV1EnvelopeService.ts

```diff
@@ -0,0 +1,85 @@
+import type { PackMessageParams } from '..'
+import type { Logger } from '../../../../../logger'
+import type { AgentContext } from '../../../../context'
+import type { DecryptedMessageContext, EncryptedMessage, SignedMessage } from '../../../types'
+import type { DIDCommV1Message } from '../DIDCommV1Message'
+
+import { DIDCommV1EnvelopeService } from '..'
+import { InjectionSymbols } from '../../../../../constants'
+import { Key, KeyType } from '../../../../../crypto'
+import { AriesFrameworkError } from '../../../../../error/AriesFrameworkError'
+import { ForwardMessage } from '../../../../../modules/routing/messages'
+import { inject, injectable } from '../../../../../plugins'
+import { Wallet } from '../../../../../wallet/Wallet'
+import { AgentConfig } from '../../../../AgentConfig'
+import { DIDCommMessageVersion, MessageType } from '../../../types'
+
+@injectable()
+export class DIDCommV1IndyEnvelopeService implements DIDCommV1EnvelopeService {
```

Comment:
I don't think this is specific to Indy right? The packing/unpacking is done using the wallet, and that could be implemented using a different library completely and still work. Maybe we can call it DidCommV1GenericEnvelopeService or something? not sure tbh